src/vmware-template.xml

```diff
@@ -0,0 +1,96 @@
+        <rasd:InstanceID>1</rasd:InstanceID>
+        <rasd:ResourceType>3</rasd:ResourceType>
+	<rasd:VirtualQuantity>{ovf_cpu_count}</rasd:VirtualQuantity>
+      </Item>
+      <Item>
+        <rasd:AllocationUnits>byte * 2^20</rasd:AllocationUnits>
+        <rasd:Description>Memory Size</rasd:Description>
+        <rasd:ElementName>4096 MB of memory</rasd:ElementName>
+        <rasd:InstanceID>2</rasd:InstanceID>
+        <rasd:ResourceType>4</rasd:ResourceType>
+	<rasd:VirtualQuantity>{ovf_memory_mb}</rasd:VirtualQuantity>
+      </Item>
+      <Item ovf:required="false">
+        <rasd:AutomaticAllocation>false</rasd:AutomaticAllocation>
+        <rasd:ElementName>VirtualVMCIDevice</rasd:ElementName>
+        <rasd:InstanceID>7</rasd:InstanceID>
+        <rasd:ResourceSubType>vmware.vmci</rasd:ResourceSubType>
+        <rasd:ResourceType>1</rasd:ResourceType>
+        <vmw:Config ovf:required="false" vmw:key="allowUnrestrictedCommunication" vmw:value="false"/>
+      </Item>
+      <Item>
+        <rasd:Address>0</rasd:Address>
+        <rasd:Description>SCSI Controller</rasd:Description>
+        <rasd:ElementName>SCSI Controller 0</rasd:ElementName>
+        <rasd:InstanceID>3</rasd:InstanceID>
+	<rasd:ResourceSubType>{vsphere_scsi_controller_type}</rasd:ResourceSubType>
+        <rasd:ResourceType>6</rasd:ResourceType>
+      </Item>
+      <Item>
+        <rasd:AddressOnParent>0</rasd:AddressOnParent>
+        <rasd:ElementName>Hard disk 0</rasd:ElementName>
+        <rasd:HostResource>ovf:/disk/vmdisk1</rasd:HostResource>
+        <rasd:InstanceID>4</rasd:InstanceID>
+        <rasd:Parent>3</rasd:Parent>
+        <rasd:ResourceType>17</rasd:ResourceType>
+        <vmw:Config ovf:required="false" vmw:key="backing.writeThrough" vmw:value="false"/>
+      </Item>
+      <Item>
+        <rasd:AddressOnParent>7</rasd:AddressOnParent>
+        <rasd:AutomaticAllocation>true</rasd:AutomaticAllocation>
+        <rasd:Connection>VM Network</rasd:Connection>
+        <rasd:Description>VmxNet3 ethernet adapter on "VM Network"</rasd:Description>
+        <rasd:ElementName>Network adapter 1</rasd:ElementName>
+        <rasd:InstanceID>5</rasd:InstanceID>
+	<rasd:ResourceSubType>{vsphere_network_controller_type}</rasd:ResourceSubType>
+        <rasd:ResourceType>10</rasd:ResourceType>
+        <vmw:Config ovf:required="false" vmw:key="connectable.allowGuestControl" vmw:value="true"/>
+        <vmw:Config ovf:required="false" vmw:key="wakeOnLanEnabled" vmw:value="false"/>
+      </Item>
+    </VirtualHardwareSection>
+    <ProductSection>
+      <Info>Information about the installed software</Info>
+      <Product>{vsphere_product_name}</Product>
+      <Vendor>{vsphere_product_vendor_name}</Vendor>
+      <Version>{vsphere_product_version}</Version>
+      <Property ovf:userConfigurable="true" ovf:type="string"
+                ovf:key="guestinfo.coreos.config.data" ovf:value="">
+        <Label>CoreOS config data</Label>
```

Comment:
I just discovered that if the OVF XML contains configurable variables for things like "guestinfo.interface.0.ip.0.address" then terraforms vsphere VM resource (vapp section) can be used to customize the VM at run time. 

https://www.terraform.io/docs/providers/vsphere/r/virtual_machine.html#using-vapp-properties-to-supply-ovf-ova-configuration

If you extract the contents of https://stable.release.core-os.net/amd64-usr/current/coreos_production_vmware_ova.ova you should see an example of the XML that we could mirror for the RHCOS deployment.

I just discovered that if the OVF XML contains configurable variables for things like "guestinfo.interface.0.ip.0.address" then terraforms vsphere VM resource (vapp section) can be used to customize the VM at run time. 

https://www.terraform.io/docs/providers/vsphere/r/virtual_machine.html#using-vapp-properties-to-supply-ovf-ova-configuration

If you extract the contents of https://stable.release.core-os.net/amd64-usr/current/coreos_production_vmware_ova.ova you should see an example of the XML that we could mirror for the RHCOS deployment.

Right now with this image the only customizable field is guest info.coreos.config.data

Reply:
Interesting. That appears to be the interface we designed for coreos-cloudinit. Nothing in RHCOS will respect those guest properties though since we use Ignition instead.

Reply:
@crawford Ignition reads that too:
https://github.com/coreos/ignition/blob/spec2x/internal/providers/vmware/vmware_amd64.go#L71

Reply:
I concur with @crawford and do believe that single-split-properties are specific to coreos-cloudinit, and coming from https://github.com/coreos/coreos-cloudinit/blob/master/Documentation/vmware-guestinfo.md.

I've opened a ticket for terraform docs to clarify and possibly drop the examples: https://github.com/terraform-providers/terraform-provider-vsphere/issues/725

While we wait for docs clarification, I suggest not to add those properties here for now.

Reply:
so we should delete the `<Property` section that refers to `guestinfo.coreos.config.data` ?

Reply:
`guestinfo.coreos.config.data` & `guestinfo.coreos.config.data.encoding` are both used by Ignition. There are other variables that are present in the CL OVF template that are `coreos-cloudinit` specific. I believe @lucab is specifically talking about those `coreos-cloudinit` specific sections not the two used by Ignition.

Reply:
@arithx so does anything need to be done here, then? 

Also, for #future should these be changed to `guestinfo.ignition.config.data` ?

Reply:
Agree with luca and crawford.  What is in there is what we can support and what we need for now.  Alex actually had me remove a few things for clarity and to avoid confusion.

Reply:
In Ignition 3.0.0 it is renamed to `guestinfo.ignition.config.data`, I believe in the `spec2x` branch both are supported.

Reply:
ok we can update that soon too once we've got a new release out